scripts/west_commands/runners/pyocd.py

```diff
@@ -144,25 +149,29 @@ def do_run(self, command, **kwargs):
             self.debug_debugserver(command, **kwargs)
 
     def flash(self, **kwargs):
-        if self.hex_name is not None and os.path.isfile(self.hex_name):
+        fformat_args = []    
+        if self.flash_format == 'hex':
             fname = self.hex_name
-        elif self.bin_name is not None and os.path.isfile(self.bin_name):
-            self.logger.warning(
-                'hex file ({}) does not exist; falling back on .bin ({}). '.
-                format(self.hex_name, self.bin_name) +
-                'Consider enabling CONFIG_BUILD_OUTPUT_HEX.')
+            fformat_args = ['--format', 'hex']
+        elif self.flash_format == 'bin':
             fname = self.bin_name
+            fformat_args = ['--format', 'bin']
+        elif self.flash_format == 'elf':
+            fname = self.elf_name
+            fformat_args = ['--format', 'elf']
         else:
+            fname = self.elf_name
```

Comment:
I must have missed this commit the first time I reviewed this PR, sorry.

This changes the default file format from hex to elf. That is a logical change which is separate from adding a new `--flash-format` option and looks unrelated to the WSL change. I have also been trying to get away from adding new runner options that basically just wrap an option to the underlying tool.

Do you mind splitting this commit into its own PR and structuring it a bit more separately into logical commits that only change one thing at a time? I want to get the WSL change merged as soon as we can, and I think this flash format change is unrelated and deserves more discussion.

Reply:
The previous code checked the file paths individually which made it more complicated to modify the paths for WSL.  I modified it so I only had to change the path once before calling it.  It also made sense to add support for .elf files, but it is simple to change the default back to .hex